exporter/elasticsearchexporter/internal/serializer/otelserializer/otelserializer.go

```diff
@@ -3,9 +3,65 @@
 
 package otelserializer // import "github.com/open-telemetry/opentelemetry-collector-contrib/exporter/elasticsearchexporter/internal/serializer/otelserializer"
 
-type Serializer struct{}
+import (
+	"fmt"
+	"time"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/exporter/elasticsearchexporter/internal/lru"
+)
+
+const (
+	knownExecutablesCacheSize             = 128 * 1024
+	knownFramesCacheSize                  = 128 * 1024
+	knownTracesCacheSize                  = 128 * 1024
+	knownUnsymbolizedFramesCacheSize      = 128 * 1024
+	knownUnsymbolizedExecutablesCacheSize = 128 * 1024
+
+	minILMRolloverTime = 3 * time.Hour
+)
+
+type Serializer struct {
+	// Data cache for profiles
+	knownTraces                  *lru.LRUSet
+	knownFrames                  *lru.LRUSet
+	knownExecutables             *lru.LRUSet
+	knownUnsymbolizedFrames      *lru.LRUSet
+	knownUnsymbolizedExecutables *lru.LRUSet
+}
 
 // New builds a new Serializer
-func New() *Serializer {
-	return &Serializer{}
+func New() (*Serializer, error) {
+	// Create LRUs with MinILMRolloverTime as lifetime to avoid losing data by ILM roll-over.
+	knownTraces, err := lru.NewLRUSet(knownTracesCacheSize, minILMRolloverTime)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create traces LRU: %w", err)
+	}
+
+	knownFrames, err := lru.NewLRUSet(knownFramesCacheSize, minILMRolloverTime)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create frames LRU: %w", err)
+	}
+
+	knownExecutables, err := lru.NewLRUSet(knownExecutablesCacheSize, minILMRolloverTime)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create executables LRU: %w", err)
+	}
+
+	knownUnsymbolizedFrames, err := lru.NewLRUSet(knownUnsymbolizedFramesCacheSize, minILMRolloverTime)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create unsymbolized frames LRU: %w", err)
+	}
+
+	knownUnsymbolizedExecutables, err := lru.NewLRUSet(knownUnsymbolizedExecutablesCacheSize, minILMRolloverTime)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create unsymbolized executables LRU: %w", err)
+	}
+
```

Comment:
We should have a go routine that triggers every 1 or 2 minutes and calls PurgeExpired() on every cache to allow the GC to clean up unused items.